experimental/list_formatter/src/list_formatter.rs

```diff
@@ -43,16 +41,18 @@ impl ListFormatter {
     }
 
     /// Returns a `Writeable` composed of the input `Writeable`s and the language-dependent
-    /// formatting. The first layer of fields contains `ListFormatter::element()` for input elements,
-    /// and `ListFormatter::literal()` for list literals.
-    pub fn format<'a, W: Writeable, B: Borrow<W>, I: IntoIterator<Item = B> + Copy>(
-        &'a self,
+    /// formatting. The first layer of fields contains `ListFormatter::element()` for input
+    /// elements, and `ListFormatter::literal()` for list literals.
+    pub fn format<'a, 'b: 'a, 'c: 'a, W: Writeable + 'b, I: IntoIterator<Item = W>>(
```

Comment:
nit: I don't think `'b` and `'c` need to be different here

Reply:
also, given that we `.into_iter()` this, why not directly accept an iterator?

Reply:
It will need `I: 'b`

Reply:
I assume the majority of usages will be on slices, and it's annoying to always add `.iter()`.

Reply:
Does it hurt to have different `'b` and `'c`? I'm a bit scared that some client will run into errors because I collapse lifetimes that should be separate.

Reply:
It can get confusing, and I don't think it's something we need to worry about here because of the variance of the lifetime usage.

I guess IntoIterator is strictly more general so it should be fine. I was mostly worried about taking ownership when unnecessary.

Reply:
updated